raiden/transfer/mediated_transfer/state.py

```diff
@@ -628,8 +628,13 @@ def __init__(
             initiator: InitiatorAddress,
             target: TargetAddress,
             secret: Secret,
+            secret_hash: SecretHash = None,
     ) -> None:
-        secrethash = sha3(secret)
+        if secret is None:
```

Comment:
Hm why is this? When can the secret be None here? I would rather this logic be moved out of here and onto wherever `TransferDescriptionWithSecretState` is populated. Also type of `Secret` in the function's args is wrong if you keep this.

Reply:
When secret-hash is injected without a secret :-) See the PR title.

Moved up  as you suggested.